main/src/hadd.cxx

```diff
@@ -119,252 +134,335 @@
+      // interpret anything after the flag as the argument.
+      nxtArg = arg + flagLen;
+      // Ignore one '=', if present
+      if (nxtArg[0] && nxtArg[0] == '=')
+         ++nxtArg;
+   } else if (argLen == flagLen) {
+      if (argIdx + 1 < argc) {
+         nxtArg = argv[argIdx + 1];
+      } else {
+         std::cerr << "[err] Expected argument after '-" << flagStr << "' flag.\n";
+         return EFlagResult::Error;
+      }
+   } else {
+      return EFlagResult::Ignored;
+   }
+
+   flagOut = conv(nxtArg);
+   ++argIdxInOut;
+   return EFlagResult::Parsed;
+}
+
+static bool ValidCompressionSettings(int compSettings)
+{
+   for (int alg = 0; alg <= 5; ++alg) {
+      for (int j=0; j<=9; ++j) {
+         int comp = (alg*100)+j;
+         if (compSettings == comp)
+            return true;
+      }
+   }
+   return false;
+}
+
+// The -f flag has a somewhat complicated logic.
+// We have 4 cases:
+//   1. -f
+//   2. -ff
+//   3. -fk
+//   4. -f[0-509]
+//
+// and a combination thereof (e.g. -fk101, -ff202, -ffk, -fk209)
+// -ff and -f[0-509] are incompatible.
+//
+// ALL these flags imply '-f' ("force overwrite"), but only if they parse successfully.
+// This means that if we see a -f[something] and that "something" doesn't parse to a valid
+// number between 0 and 509, or f or k, we consider the flag invalid and skip it without
+// setting any state.
+//
+// Note that we don't allow `-f [0-9]` because that would be a backwards-incompatible
+// change with the previous arg parsing semantic, changing the meaning of a cmdline like:
+//
+// $ hadd -f 200 f.root g.root  # <- '200' is the output file, not an argument to -f!
+static EFlagResult FlagF(const char *arg, HaddArgs &args)
+{
+   if (arg[0] != 'f')
+      return EFlagResult::Ignored;
+
+   args.fForce = true;
+   const char *cur = arg + 1;
+   while (*cur) {
+      switch (cur[0]) {
+      case 'f':
+         if (args.fUseFirstInputCompression)
+            std::cerr << "[warn] Duplicate flag: -ff\n";
+         if (args.fCompressionSettings) {
+            std::cerr << "[err] Cannot specify both -ff and -f[0-9]. Either use the first input compression or specify it.\n";
+            return EFlagResult::Error;
+         } else
+            args.fUseFirstInputCompression = true;
+         break;
+      case 'k':
+         if (args.fKeepCompressionAsIs)
+            std::cerr << "[warn] Duplicate flag: -fk\n";
+         args.fKeepCompressionAsIs = true;
+         break;
+      default:
+         if (isdigit(cur[0])) {
+            if (args.fUseFirstInputCompression) {
+               std::cerr << "[err] Cannot specify both -ff and -f[0-9]. Either use the first input compression or specify it.\n";
+               return EFlagResult::Error;
+            } else if (!args.fCompressionSettings) {
+               if (auto compLv = StrToInt(cur)) {
+                  if (ValidCompressionSettings(*compLv)) {
+                     args.fCompressionSettings = *compLv;
```

Comment:
Should we return immediately here? It has to be the last digits of the `-f...` arg, I think.

Reply:
Since `StrToInt` parses all the remaining chars of `arg`, we will break out of the while at the end of this loop. If there are trailing characters after the digits `StrToInt` will error out